File: Samples/Samples/MenuSamples.cs
// 
// MenuSamples.cs
//  
// Author:
//       Lluis Sanchez <lluis@xamarin.com>
// 
// Copyright (c) 2012 Xamarin Inc
// 
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
// 
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
// 
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
// THE SOFTWARE.
using System;
using Xwt;

namespace Samples
{
	public class MenuSamples: VBox
	{
		Menu menu;
		
		public MenuSamples ()
		{
			Label la = new Label ("Right click here to show the context menu");
			menu = new Menu ();
			menu.Items.Add (new MenuItem ("One"));
			menu.Items.Add (new MenuItem ("Two"));
			menu.Items.Add (new MenuItem ("Three"));
			
			la.ButtonPressed += HandleButtonPressed;
			PackStart (la);
		}

		void HandleButtonPressed (object sender, ButtonEventArgs e)
		{
			if (e.Button == PointerButton.Right)
				menu.Popup ();
		}
	}
}

File: SamplesVB/Samples/MenuSamples.vb
Imports System
Imports Xwt

Namespace Samples
    Public Class MenuSamples
        Inherits VBox

        Private menu As Menu

        Public Sub New()
            Dim la As Label = New Label("Right click here to show the context menu")
            Me.menu = New Menu()
            Me.menu.Items.Add(New MenuItem("One"))
            Me.menu.Items.Add(New MenuItem("Two"))
            Me.menu.Items.Add(New MenuItem("Three"))
            AddHandler la.ButtonPressed, AddressOf Me.HandleButtonPressed
            MyBase.PackStart(la)
        End Sub

        Private Sub HandleButtonPressed(sender As Object, e As ButtonEventArgs)
            If e.Button = PointerButton.Right Then
                Me.menu.Popup()
            End If
        End Sub
    End Class
End Namespace
